Files/View Models/Properties/FileProperty.cs

```diff
@@ -0,0 +1,259 @@
+            }
+
+            if (Value != null && Value.GetType().IsArray)
+            {
+                if (Value.GetType().GetElementType().Equals(typeof(string)))
+                {
+                    return new StringArrayToString();
+                }
+
+                if (Value.GetType().GetElementType().Equals(typeof(double)))
+                {
+                    return new DoubleArrayToString();
+                }
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Converts the property to a string to be displayed
+        /// </summary>
+        /// <returns></returns>
+        private string ConvertToString()
+        {
+            if (DisplayFunction != null)
+            {
+                return DisplayFunction.Invoke(Value);
+            }
+
+            if (Converter != null && Value != null)
+            {
+                return Converter.Convert(Value, typeof(string), null, null) as string;
+            }
+
+            return Value as string;
+        }
+
+        /// <summary>
+        /// Converts a string from a text block back to it's original type
+        /// </summary>
+        /// <param name="value">The input string</param>
+        /// <returns></returns> 
+        private object ConvertBack(string value)
+        {
+            if (Converter != null && value != null)
+            {
+                return Converter.ConvertBack(value, typeof(object), null, null);
+            }
+
+            return value;
+        }
+
+        public async static Task<List<FileProperty>> RetrieveAndInitializePropertiesAsync(StorageFile file)
+        {
+            var propertiesJsonFile = await StorageFile.GetFileFromApplicationUriAsync(new Uri(@"ms-appx:///Resources/PropertiesInformation.json"));
+            var list = JsonConvert.DeserializeObject<List<FileProperty>>(await FileIO.ReadTextAsync(propertiesJsonFile));
+
+            var propsToGet = new List<string>();
+
+            foreach (var prop in list)
+            {
+                if (!string.IsNullOrEmpty(prop.Property))
+                {
+                    propsToGet.Add(prop.Property);
+                }
+            }
+
+            var keyValuePairs = await file.Properties.RetrievePropertiesAsync(propsToGet);
+
+            foreach (var prop in list)
+            {
+                if (!string.IsNullOrEmpty(prop.Property))
+                {
+                    prop.Value = keyValuePairs[prop.Property];
+                }
+
+                prop.InitializeProperty();
+            }
+
+            return list;
+        }
+
+        /// <summary>
+        /// Since you can't serialize lambdas from a json file, define them here
+        /// </summary>
+        private static readonly Dictionary<string, Func<object, string>> DisplayFuncs = new Dictionary<string, Func<object, string>>()
+        {
+            { "DivideBy1000", input => (((uint) input)/1000).ToString() },
```

Comment:
I guess here should be added null check.

Reply:
@tsvietOK Good point, though I think it might be simpler to add a null check in the ConvertToString function, so that no null values get passed to any converter.